be2-scala/src/main/scala/ch/epfl/pop/model/objects/LaoData.scala

```diff
@@ -18,23 +18,25 @@ import com.google.crypto.tink.subtle.Ed25519Sign
  * @param privateKey : the LAO's own private key, used to sign messages
  * @param publicKey  : the LAO's own public key, used to sign messages
  * @param witnesses  : the LAO's list of witnesses
+ * @param address : the canonical address of the Server
  */
 final case class LaoData(
                           owner: PublicKey,
                           attendees: List[PublicKey],
                           privateKey: PrivateKey,
                           publicKey: PublicKey,
-                          witnesses: List[PublicKey]
+                          witnesses: List[PublicKey],
+                          address: String
                         ) {
   def toJsonString: String = {
     val that: LaoData = this // tricks the compiler into inferring the right type
     that.toJson.toString
   }
 
-  def updateWith(message: Message): LaoData = {
+  def updateWith(message: Message, address: String): LaoData = {
```

Comment:
referring to comments below, could be cleaner with an Option

Reply:
Also, I don't know whether this could be an issue, but currently, you will not be able to change the address unless the associated message is a CloseRollCall or CreateLao (from what I understood, it should only happen during Lao creation so it would not be a problem but still wanted to point this out)

Reply:
Yes it is true, but as you said it sould not be changed in other handlers except in the lao, since it is where the greeting message is sent. So I thought that it should stay as it was: only concerning the rollcall and election